src/tpke_pubkey.erl

```diff
@@ -104,14 +116,15 @@ verify_signature(PubKey, Signature, H) ->
     B = erlang_pbc:element_pairing(H, PubKey#pubkey.verification_key),
     erlang_pbc:element_cmp(A, B).
 
--spec combine_signature_shares(pubkey(), [tpke_privkey:share(), ...]) -> erlang_pbc:element().
-combine_signature_shares(PubKey, Shares) ->
+-spec combine_signature_shares(pubkey(), [tpke_privkey:share(), ...], ciphertext()) -> erlang_pbc:element().
+combine_signature_shares(PubKey, Shares, {U, V, W}) ->
     {Indices, _} = lists:unzip(Shares),
     Set = ordsets:from_list(Indices),
     MySet = ordsets:from_list(lists:seq(0, PubKey#pubkey.players - 1)),
     true = ordsets:is_subset(Set, MySet),
 
-    %% TODO for robustness we should verify each share before combining them
+    %% for robustness we verify each share before combining them
+    lists:all(fun({Index, Share}) -> verify_share(PubKey, {Index, Share}, {U, V, W}) end, Shares),
```

Comment:
You don't use the result of this test, and you should be using 'verify_signature_share' here.

Reply:
updated in a743393